ethers-signers/src/wallet/mod.rs

```diff
@@ -71,12 +71,21 @@ pub struct Wallet<D: DigestSigner<Sha256Proxy, RecoverableSignature>> {
     pub(crate) address: Address,
     /// The wallet's chain id (for EIP-155)
     pub(crate) chain_id: u64,
+    // The wallet's mnemonic
+    pub mnemonic: Option<String>,
+    // The wallet's private key
+    pub private_key: Option<String>,
+    // The wallet's public key
+    pub public_key: Option<String>
```

Comment:
I don't think this is a good idea.
the `Wallet` only requires that the signer type is able to sign data and the signer may not even have access to the private key directly.
and if it does (LocalWallet) then private_key/public_key is redundant here.

Making them public also makes it possible to modify them which could cause all sorts of issues.

Reply:
Thank you for the feedback.

Some things to think about with this pull request:

1. I haven't fundamentally changed the way wallets work. If you change these values all that will happen is the these values will change. They are never actually used anywhere. I appreciate they could be in the future and making them constants would probably help remediate this.

2. I'm not sure if I have misunderstood your concern about leaking these values via println but two things I would think of here are that if someone can modify the library then they could just make the changes I made and if you're concerned about someone implementing this accidentally leaking a private key via println could they not also leak the same data via the method I used to assign the private key variable just as easily? (`format!("0x{:02X?}", signer.to_bytes()).replace(", ", "").replace("[", "").replace("]", "")`)